### PR TITLE
Feature/net bcast

### PR DESCRIPTION
**Pull request by prekucki**
* bcast api for net
* from/to sending for net

### CODE REVIEW

File: core/net/src/service.rs [hunk 1]
```diff
@@ -19,27 +23,44 @@ pub fn central_net_addr() -> std::io::Result<SocketAddr> {
         .expect("central net hub addr needed"))
 }
 
+#[inline]
+fn net_node_id(node_id: &NodeId) -> String {
+    format!("{}/{:?}", net::BUS_ID, node_id)
+}
+
+fn parse_from_addr(from_addr: &str) -> anyhow::Result<(NodeId, NodeId, &str)> {
```
Comment (Wiezzel) [inline core/net/src/service.rs:31]: Could de done easier with a regex: `/from/(\w+)/to/(\w+)(/[\w/]+)`

File: core/net/src/api.rs [hunk 2]
```diff
@@ -31,8 +31,52 @@ impl TryRemoteEndpoint for &str {
...
+    pub fn to(&self, dst: NodeId) -> NetDst {
+        NetDst {
+            identity: self.identity,
+            dst,
+        }
+    }
+}
+
+pub struct NetDst {
+    identity: NodeId,
```
Comment (Wiezzel) [inline core/net/src/api.rs:48]: Both fields are identities. I suggest to rename `identity` to `src`.

File: core/net/src/service.rs (hunk 1 above)
Comment (prekucki) [reply core/net/src/service.rs:31]: it won't:

- regex compilation runs in runtime. so i need lazy static or something.

does not looks very simpler?. and it is impossible to return borrowed slice from an argument. (or i do not know how to do this). so it has additional allocations on heap.

```rust
fn parse_from_addr(from_addr: &str) -> anyhow::Result<(NodeId, NodeId, String)> {
    let re = regex::Regex::new("/from/(\\w+)/to/(\\w+)(/[\\w/]+)")?;
    if let Some(c) = re.captures(from_addr) {
        let (from_node_id, to_node_id, service_id) = (&c[1], &c[2], &c[3]);
        return Ok((from_node_id.parse()?, to_node_id.parse()?, service_id.to_owned()));
    }
    Err(anyhow!("invalid net-from destination: {}", from_addr))
}
```

File: core/net/src/api.rs (hunk 2 above)
Comment (prekucki) [reply core/net/src/api.rs:48]: i was thinking about `src` when i wrote this. I was thinking about identity selection for the caller.

File: core/net/src/service.rs (hunk 1 above)
Comment (Wiezzel) [reply core/net/src/service.rs:31]: It's clearer to see what the function does, but if it's too much trouble then never mind.

**Pull request merged**